### ISSUE TITLE
Bug: no-obj-calls reports unavailable globals via global object

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#20674]
### Environment

Node version: v24.14.0
npm version: v11.9.0
Local ESLint version: v10.1.0
Global ESLint version:
Operating System:

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    rules: {
      "no-obj-calls": "error",
    },
    languageOptions: {
      ecmaVersion: 6,
      globals: {
        window: false,
      }
    },
  },
];
```
</details>

```js
window.Atomics();
```

### What did you expect to happen?

I expected `no-obj-calls` not to report an error.

`Atomics` is not available at `ecmaVersion: 6`, so `window.Atomics()` should be treated as a normal property access on `window`, not as a call to the built-in `Atomics` global.

### What actually happened?

`no-obj-calls` reports:
```
'Atomics' is not a function.
```

### Link to Minimal Reproducible Example

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZ2xvYmFsIHdpbmRvdyAqL1xud2luZG93LkF0b21pY3MoKTtcbkF0b21pY3MoKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLW9iai1jYWxscyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX0sImVjbWFWZXJzaW9uIjoyMDE1fX19)

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I agree we should not assume that `window.GLOBAL` and `GLOBAL` refer to the same object when the `GLOBAL` is unset.
This problem seems to originate from the `ReferenceTracker` (which [multiple other rules](https://github.com/search?q=repo%3Aeslint%2Feslint%20ReferenceTracker&type=code)) also use.
So I think we should rather report this problem upstream to [`@eslint-community/eslint-utils`](https://github.com/eslint-community/eslint-utils) than fix it here.

**Comment by snitin315** [comment]
Agree with @DMartens, it should be fixed upstream. Marking this accepted for tracking until fixed in [@eslint-community/eslint-utils](https://github.com/eslint-community/eslint-utils)

**Comment by Kuldeep2822k** [comment]
is someone currently work on this issue if not i be happy to solve it

**Comment by lumirlumir** [comment]
@Kuldeep2822k Thanks for your interest in contributing, but this issue is currently blocked and already has an assignee who is willing to submit a PR.
